src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,225 @@
+import duke.command.CommandRecipeIngredient;
+import duke.command.CommandRecipeTitle;
+import duke.exception.DukeException;
+import duke.list.bookinglist.BookingList;
+import duke.list.ingredientlist.IngredientList;
+import duke.list.recipelist.RecipeIngredientList;
+import duke.parser.Parser;
+import duke.list.recipelist.RecipeTitleList;
+import duke.storage.BookingStorage;
+import duke.storage.IngredientStorage;
+import duke.storage.RecipeIngredientStorage;
+import duke.storage.RecipeTitleStorage;
+import duke.task.recipetasks.Feedback;
+import duke.task.recipetasks.Rating;
+import duke.task.recipetasks.RecipeIngredient;
+import duke.ui.Ui;
+
+import java.util.ArrayList;
+import java.text.ParseException;
+
+import static duke.common.BookingMessages.*;
+import static duke.common.IngredientMessages.*;
+import static duke.common.Messages.*;
+import static duke.common.RecipeMessages.*;
+
+/**
+ * Duke processes different commands.
+ */
+public class Duke {
+
+    private Ui ui;
+
+    private IngredientStorage ingredientStorage;
+    private RecipeIngredientStorage recipeIngredientStorage;
+    private BookingStorage bookingStorage;
+    private RecipeTitleStorage recipeTitleStorage;
+    private IngredientList ingredientList;
+    private RecipeIngredientList recipeIngredientList;
+    private BookingList bookingList;
+    private RecipeTitleList recipeTitleList;
+    private RecipeIngredient recipeIngredient;
+    private Rating rating;
+    private Feedback feedback;
+
+//    /**
+//     * Constructor for Duke class to instantiation Ui, Storage, TaskList classes.
+//     * @param filePath String containing the directory in which the tasks are to be stored
+//     */
+
+    public Duke(Ui ui) {
+        this.ui = ui;
+        ingredientStorage = new IngredientStorage(filePathIngredients);
+        recipeIngredientStorage = new RecipeIngredientStorage(filePathRecipeIngredients);
+        recipeTitleStorage = new RecipeTitleStorage(filePathRecipeTitle);
+        bookingStorage = new BookingStorage(filePathBookings);
+
+        try {
+            ingredientList = new IngredientList(ingredientStorage.load());
+            recipeIngredientList = new RecipeIngredientList(recipeIngredientStorage.load());
+            recipeTitleList = new RecipeTitleList(recipeTitleStorage.load());
+            bookingList = new BookingList(bookingStorage.load());
+        } catch (DukeException e) {
+            ui.showIngredientLoadingError();
+            ui.showLoadingError();
+        }
+    }
+
+    public String showWelcome() {
+        return ui.showWelcome();
+    }
+
+    public ArrayList<String> runProgram(String userInput) throws DukeException, ParseException {
+        System.out.println("stuck here1");
+        ArrayList<String> arrayList = new ArrayList<>();
+        if (userInput.contains(COMMAND_ADD_RECIPE_TITLE)) {
+            System.out.println("stuck here2");
+            if (userInput.trim().substring(0, 14).equals(COMMAND_ADD_RECIPE_TITLE)) {
```

Comment:
Should the runProgram method deals with string manipulations? I hope its responsibility is to call relevant objects to initiate the run